### ISSUE TITLE
DHCP server does not seem to handle option 26

### ISSUE THREAD

**Issue by sjorge** [opnsense/core#1565]
Looks like it is not possible to send option 26 to specify a MTU.
My storage network uses an MTU of 9000.

Interface MTU Option (https://tools.ietf.org/html/rfc2132)
![sceenshot](http://i.imgur.com/TsiBF2w.png)

```
option domain-name "acheron.be";
option ldap-server code 95 = text;
option domain-search-list code 119 = text;
option arch code 93 = unsigned integer 16; # RFC4578
option custom-opt2-0 code 26 = unsigned integer 16;

default-lease-time 7200;
max-lease-time 86400;
log-facility local7;
one-lease-per-client true;
deny duplicates;
ping-check true;
update-conflict-detection false;
authoritative;
failover peer "dhcp_lan" {
  secondary;
  address 172.16.40.12;
  port 520;
  peer address 172.16.40.11;
  peer port 519;
  max-response-delay 10;
  max-unacked-updates 10;

  load balance max seconds 3;
}

failover peer "dhcp_opt3" {
  secondary;
  address 172.16.10.12;
  port 520;
  peer address 172.16.10.11;
  peer port 519;
  max-response-delay 10;
  max-unacked-updates 10;

  load balance max seconds 3;
}

failover peer "dhcp_opt1" {
  secondary;
  address 172.16.30.12;
  port 520;
  peer address 172.16.30.11;
  peer port 519;
  max-response-delay 10;
  max-unacked-updates 10;

  load balance max seconds 3;
}

failover peer "dhcp_opt2" {
  secondary;
  address 172.16.20.12;
  port 520;
  peer address 172.16.20.11;
  peer port 519;
  max-response-delay 10;
  max-unacked-updates 10;

  load balance max seconds 3;
}


subnet 172.16.40.0 netmask 255.255.255.0 {
  pool {
    option domain-name-servers 172.16.40.1;
    deny dynamic bootp clients;
    failover peer "dhcp_lan";
    range 172.16.40.245 172.16.40.250;
  }

  option routers 172.16.40.1;
  option domain-name "vlan40.acheron.be";
  option domain-search "acheron.be";
  option domain-name-servers 172.16.40.1;
  default-lease-time 86400;
}

subnet 172.16.10.0 netmask 255.255.255.0 {
  pool {
    option domain-name-servers 172.16.10.1;
    deny dynamic bootp clients;
    failover peer "dhcp_opt3";
    range 172.16.10.225 172.16.10.250;
  }

  option routers 172.16.10.1;
  option domain-name "vlan10.acheron.be";
  option domain-search "acheron.be";
  option domain-name-servers 172.16.10.1;
  default-lease-time 3600;
}

subnet 172.16.30.0 netmask 255.255.255.0 {
  pool {
    option domain-name-servers 172.16.30.1;
    deny dynamic bootp clients;
    failover peer "dhcp_opt1";
    range 172.16.30.245 172.16.30.250;
  }

  option routers 172.16.30.1;
  option domain-name "vlan30.acheron.be";
  option domain-search "acheron.be";
  option domain-name-servers 172.16.30.1;
  default-lease-time 86400;
}

subnet 172.16.20.0 netmask 255.255.255.0 {
  pool {
    option domain-name-servers 172.16.20.1;
    deny dynamic bootp clients;
    failover peer "dhcp_opt2";
    range 172.16.20.245 172.16.20.250;
  }

  option routers 172.16.20.1;
  option domain-name "vlan20.example.org";
  option domain-search "acheron.be";
  option domain-name-servers 172.16.20.1;
  default-lease-time 86400;

  option custom-opt2-0 9000;
}
```

I was expecting ```option interface-mtu 9000``` to appear an not a custom option.
The above does not seem to work for me either.

**Comment by sjorge** [comment]
There should probably be some filtering to grab the options that are already defined in dhcpd-options

**Comment by AdSchellevis** [comment]
Hi Jorge,

I don't think there's a switch to list all available options and their  names for dhcpd, otherwise it would be a good idea to auto rename the options when possible.
Simplest solution in this case is add the option like the others to the page it self.

Best regards,

Ad

**Comment by sjorge** [comment]
If I am not mistaken the dhpcd-options man page lists them.
Although just which ones are available, not what id they map too. But they do seem to have the same name as in the rfc.

**Comment by AdSchellevis** [comment]
Yes, I know where to find them, we just can't probe them automatically. 
We can add this option to the page and align functionality with the rest, or keep this issue open for better implementation ideas.

**Comment by sjorge** [comment]
Fair enough, MTU looks like a good candidate to add as we also allow setting this on the interface configuration.

Some of the other existing ones seems very niche.

**Comment by AdSchellevis** [comment]
I agree, there certainly are better solutions for this, but they all will take more time then I have available at the moment.

**Comment by AdSchellevis** [comment]
@sjorge can you try https://github.com/opnsense/core/commit/025021b867b082e4ea27928ec6433dc21e4e78b0 ?

**Comment by sjorge** [comment]
```bash
root@exosphere:~ # opnsense-patch 025021b
Hmm...  Looks like a unified diff to me...
The text leading up to this was:
--------------------------
|From 025021b867b082e4ea27928ec6433dc21e4e78b0 Mon Sep 17 00:00:00 2001
|From: Ad Schellevis <ad@opnsense.org>
|Date: Sat, 29 Apr 2017 13:17:39 +0200
|Subject: [PATCH] dhcpd, add interface-mtu option, for
| https://github.com/opnsense/core/issues/1565
|
|---
| src/etc/inc/services.inc  |  9 +++++++++
| src/www/services_dhcp.php | 17 ++++++++++++++++-
| 2 files changed, 25 insertions(+), 1 deletion(-)
|
|diff --git a/src/etc/inc/services.inc b/src/etc/inc/services.inc
|index 9d3471c..0e3bd4e 100644
|--- a/src/etc/inc/services.inc
|+++ b/src/etc/inc/services.inc
--------------------------
Patching file etc/inc/services.inc using Plan A...
Hunk #1 succeeded at 687 (offset 11 lines).
Hunk #2 succeeded at 758 (offset 11 lines).
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git a/src/www/services_dhcp.php b/src/www/services_dhcp.php
|index 867b3ef..530e1ec 100644
|--- a/src/www/services_dhcp.php
|+++ b/src/www/services_dhcp.php
--------------------------
Patching file www/services_dhcp.php using Plan A...
Hunk #1 succeeded at 97.
Hunk #2 succeeded at 323.
Hunk #3 succeeded at 858.
done
All patches have been applied successfully.  Have a nice day.
### do the needful in the webui
root@exosphere:~ # cat /var/dhcpd/etc/dhcpd.conf | grep interface-mtu
  option interface-mtu 9000;
```

Looks good to me!

PS: opnsense-patch is pretty cool! I assume the next .1 release will include this and install as normal?

**Comment by AdSchellevis** [comment]
yes, opnsense-patch is very handy :)
let's ask @fichtner when we can expect this one in stable. Can we close this issue now?

**Comment by sjorge** [comment]
Yeah this can be closed, 17.7 would be fine, the next 17.1.x release would be better :)

**Comment by fichtner** [comment]
just backported this for 17.1.6, thanks guys 👍 

btw, running opnsense-patch again removes the patch
